dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/service/GDBProcesses_7_2.java

```diff
@@ -333,38 +327,38 @@ public void attachDebuggerToProcess(final IProcessDMContext procCtx, final Strin
 							public void execute(final RequestMonitor rm) {
 								// The remote session is already connected to the process
 								// Bug 528145
-								if (fBackend.getSessionType() == SessionType.REMOTE) {
+								if (fBackend.getSessionType() == SessionType.REMOTE
+										&& doCanDetachDebuggerFromProcess()) {
```

Comment:
I am unable to understand the reason for this condition..

I am running multi-gdbserver and attaching to second process using `GdbConnectCommand` and because of this condition, actual attach is not happening.. am I missing something here?

Reply:
@umairsair can you raise a new issue for this please so we can get it tested and resolved.

Reply:
@jonahgraham I'll try to work on it in detail and will submit a patch if there is really an issue on CDT side and not something in my local setup..

Reply:
I have created https://github.com/eclipse-cdt/cdt/pull/336